In [1]:
import pandas as pd

orders = pd.read_csv('orders DA.csv')
customers = pd.read_csv('customers DA.csv')

orders['order_date'] = pd.to_datetime(orders['order_date'], format='%d-%m-%Y')

In [2]:
gmv = orders['final_amount'].sum()

delivered_orders = orders[orders['status'] == 'Delivered']
net_revenue = delivered_orders['final_amount'].sum()

aov = net_revenue / len(delivered_orders)

print(f"Gross Merchandise Value (GMV): {gmv:,.2f}")
print(f"Net Revenue: {net_revenue:,.2f}")
print(f"Average Order Value (AOV): {aov:,.2f}")

Gross Merchandise Value (GMV): 3,147,385,544.58
Net Revenue: 2,053,858,611.96
Average Order Value (AOV): 63,197.59


In [3]:
returns_df = pd.read_csv('returns DA.csv')

cancelled_orders = orders[orders['status'] == 'Cancelled']
cancellation_rate = (len(cancelled_orders) / len(orders)) * 100

return_rate = (len(returns_df) / len(delivered_orders)) * 100

print(f"Cancellation Rate: {cancellation_rate:.2f}%")
print(f"Return Rate: {return_rate:.2f}%")

Cancellation Rate: 11.99%
Return Rate: 30.77%


In [5]:
print(payments.columns)

Index(['payment_id', 'order_id', 'customer_id', 'payment_date', 'payment_time',
       'payment_method', 'amount', 'status', 'transaction_id', 'bank_name',
       'gateway', 'refund_amount', 'refund_date'],
      dtype='str')


In [6]:
payments = pd.read_csv('payments DA.csv')

clv = customers.groupby('segment')['total_spent'].mean()

failed_payments = len(payments[payments['status'] == 'Failed'])
payment_failure_rate = (failed_payments / len(payments)) * 100

print("Customer Lifetime Value (CLV) by Segment:")
print(clv.apply(lambda x: f"Rs.{x:,.0f}"))
print(f"\nPayment Failure Rate: {payment_failure_rate:.2f}%")

Customer Lifetime Value (CLV) by Segment:
segment
Budget      Rs.186,179
Inactive     Rs.32,249
New         Rs.274,866
Premium     Rs.532,735
Regular     Rs.361,871
Name: total_spent, dtype: str

Payment Failure Rate: 3.50%


In [8]:
print(inventory.columns)

Index(['inventory_id', 'product_id', 'warehouse_location',
       'quantity_available', 'quantity_reserved', 'reorder_level',
       'reorder_quantity', 'last_restocked_date', 'unit_cost',
       'total_inventory_value', 'status'],
      dtype='str')


In [9]:
order_items = pd.read_csv('order_items DA.csv')
inventory = pd.read_csv('inventory DA.csv')

monthly_gmv = orders.groupby(orders['order_date'].dt.to_period('M'))['final_amount'].sum()
mom_growth = monthly_gmv.pct_change() * 100

cat_revenue = order_items.groupby('category')['total_price'].sum()
top_category = cat_revenue.idxmax()
top_cat_share = (cat_revenue.max() / cat_revenue.sum()) * 100

in_stock = len(inventory[inventory['quantity_available'] > 0])
inventory_fill_rate = (in_stock / len(inventory)) * 100

print(f"Top Category: {top_category} ({top_cat_share:.2f}% Revenue Share)")
print(f"Inventory Fill Rate: {inventory_fill_rate:.2f}%\n")
print("Month-over-Month Growth (Last 3 Months):")
print(mom_growth.tail(3).apply(lambda x: f"{x:.2f}%"))

Top Category: Electronics (57.41% Revenue Share)
Inventory Fill Rate: 99.80%

Month-over-Month Growth (Last 3 Months):
order_date
2025-04     -4.15%
2025-05      3.84%
2025-06    -23.74%
Freq: M, Name: final_amount, dtype: str


# Final KPI Insights

### Key Observations

* **Inventory Performance:** The Inventory Fill Rate is highly optimal at **99.80%**, successfully exceeding the benchmark target of >95%. This indicates highly efficient warehouse management with minimal out-of-stock items.
* **Revenue Concentration:** The **Electronics** category drives **57.41%** of total revenue. While it is the strongest performer, it sits just below the 60% concentration risk threshold, meaning the business is heavily reliant on this single category.
* **Growth Warning:** There is a critical **-23.74%** drop in Gross Merchandise Value (GMV) observed in June 2025. This sudden decline in sales is a major red flag that management will need to investigate and address immediately.